cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/LanguageTraits.kt

```diff
@@ -43,3 +44,68 @@ interface HasTemplates : LanguageTrait
  * functions.
  */
 interface HasDefaultArguments : LanguageTrait
+
+/**
+ * A language trait that specifies that this language has a complex call resolution that we need to
+ * fine-tune in the language implementation.
+ */
+interface HasComplexCallResolution : LanguageTrait {
+    // TODO: better name?
+    fun doBetterCallResolution()
```

Comment:
Are we using this yet? How are we handling the former `is CXXLanguageFrontend` calls now?

Reply:
We do not use it yet. The problem is that there are many methods which are Cpp specific. So, we either need multiple such "do it better" methods or to distinguish the cases in the method. We also have to check which arguments the methods need.

The current check is either `is CPPLanguage` or `is HasComplexCallResolution`.